pkg/app/launcher/cmd/launcher/launcher.go

```diff
@@ -267,12 +267,20 @@ func (l *launcher) run(ctx context.Context, input cli.Input) error {
 
 	group.Go(func() error {
 		// Execute the first time immediately.
-		execute()
+		if err := execute(); err != nil {
+			input.Logger.Error("LAUNCHER: failed while launching new Piped", zap.Error(err))
+			// Return an error if the initial startup fails.
+			return err
+		}
 
 		for {
 			select {
 			case <-ticker.C:
-				execute()
+				if err := execute(); err != nil {
+					input.Logger.Error("LAUNCHER: failed while checking whether relaunch required or relaunching. Skip the loop this time", zap.Error(err))
+					// Don't return an error to continue piped execution.
+					return nil
+				}
```

Comment:
I think, in case of want to print out the error, making it print inside the `execute()` function is better since we have the cause in detail in can print the error log more clarity.

Reply:
There already exists logging error detail inside `execute()`, So should I delete this logging?

Reply:
And this `return` will stop the execution of launcher and we don't want that, so I think it's better to keep the execution in this loop as is.

Reply:
> this return will stop the execution of launcher
I made a mistake 😱  I'll fix it.

Reply:
fixed https://github.com/pipe-cd/pipe/pull/2732/commits/3121014691cb5fd66983c30fbc20122c09dd8ddc